lms/djangoapps/teams/static/teams/js/views/team_profile.js

```diff
@@ -73,10 +73,18 @@
                     event.preventDefault();
                     var view = this;
                     $.ajax({
-                       type: 'DELETE',
-                       url: view.teamMembershipDetailUrl.replace('team_id', view.model.get('id'))
+                        type: 'DELETE',
+                        url: view.teamMembershipDetailUrl.replace('team_id', view.model.get('id'))
                     }).done(function (data) {
-                       view.model.fetch({});
+                        view.teamEvents.trigger('teams:update', {
+                            action: 'leave',
+                            team: view.model
+                        });
+                        view.model.fetch({})
+                            .done(function() {
+                                view.model.isStale = false;
```

Comment:
I'm wondering if the teamEvents trigger should happen in this "done" since the teams:update sends view.model as the team (and it won't be up-to-date until the fetch is complete).